### PR TITLE
feat(viewer): storey sort order for the model browser (#1296)

### LINKED ISSUE #1296: Browser Sort Order
Should there be a sorting option for these?  Alphanumeric?  I am assuming looking at the current order it might be sorting by IfcGUID value?

Image

On a side note, would it be possible to add model navigation controls from a 3D Spacemouse?

### PR DESCRIPTION
**Pull request by louistrue**
Closes #1296.

## Problem
The model browser sorted storeys only by elevation (highest first). When storey names do not track height, that order reads as arbitrary, and the issue asked for an alphanumeric option.

## Change
Adds a sort selector under the grouping toggle, shown only in the spatial view (the Class / Type / Material trees are already name-sorted). Four modes:

- Elevation, high to low (default, preserves prior behavior)
- Elevation, low to high
- Name, A to Z
- Name, Z to A

Name sorting uses natural numeric collation, so `Level 2` sorts before `Level 10` (not after). The choice persists in `localStorage` and applies uniformly to single-model storeys, federated unified storeys, and storey-like spatial parts (`IfcFacilityPart` etc.).

The Spacemouse navigation idea raised in the issue is a separate input-device feature and is intentionally out of scope here.

## Files
- `hierarchy/types.ts` - `HierarchySortMode` type + default
- `hierarchy/treeDataBuilder.ts` - `compareStoreyEntries` comparator threaded through `buildUnifiedStoreys` / `buildTreeData` / `buildSpatialNodes`
- `hierarchy/useHierarchyTree.ts` - sort state + `localStorage` persistence
- `hierarchy/HierarchySortControl.tsx` - new dropdown control
- `HierarchyPanel.tsx` - wires the control into both layouts
- `hierarchy/treeDataBuilder.test.ts` - 7 new tests

## Verification
- `tsc --noEmit`: clean on all touched files.
- Tests: 12/12 pass, including the natural-numeric `Level 2` before `Level 10` case and the unchanged default-order regression.

## Summary by CodeRabbit

* **New Features**
  * Added a storey sorting control to the hierarchy panel when using spatial grouping mode.
  * Users can now sort entries by elevation (high-to-low or low-to-high) or by name (A–Z or Z–A).
  * The selected sort preference is saved and automatically restored between sessions.

* **Tests**
  * Expanded test coverage for storey ordering across multiple sort modes and edge cases.

### CODE REVIEW

File: apps/viewer/src/components/viewer/hierarchy/treeDataBuilder.ts [hunk 1]
```diff
@@ -310,10 +338,12 @@ function buildSpatialNodes(
...
-    // Sort storeys by elevation descending
-    const shouldSortByElevation = (spatialNode.children || []).some((child) => isStoreyLikeSpatialType(child.type));
-    const sortedChildren = shouldSortByElevation
-      ? [...(spatialNode.children || [])].sort((a, b) => (b.elevation || 0) - (a.elevation || 0))
+    // Reorder storey-like children by the chosen browser sort mode (#1296).
+    // Only storey lists are reordered; other spatial children (Site, Building)
+    // keep their document order.
+    const containsStoreys = (spatialNode.children || []).some((child) => isStoreyLikeSpatialType(child.type));
+    const sortedChildren = containsStoreys
+      ? [...(spatialNode.children || [])].sort((a, b) => compareStoreyEntries(a, b, sortMode))
```
[the remark this answers is not in the document]
Comment (louistrue) [reply apps/viewer/src/components/viewer/hierarchy/treeDataBuilder.ts]: Fixed in db74c310. Added an `isLevelLikeSpatialType` predicate so the sort now also covers `IfcFacilityPart`/`IfcBridgePart`/`IfcRoadPart`/`IfcRailwayPart`, not just `IfcBuildingStorey`. Test added for facility-part name sorting.

File: apps/viewer/src/components/viewer/hierarchy/treeDataBuilder.ts [hunk 2]
```diff
@@ -310,10 +338,12 @@ function buildSpatialNodes(
...
       : spatialNode.children || [];
```
[the remark this answers is not in the document]
Comment (louistrue) [reply apps/viewer/src/components/viewer/hierarchy/treeDataBuilder.ts]: Fixed in db74c310. Only the level-like rows are now sorted in place; non-level siblings (Site / Building / Space) keep their original document position. Test added to lock this in.

File: apps/viewer/src/components/viewer/hierarchy/useHierarchyTree.ts [hunk 3]
```diff
@@ -21,6 +22,18 @@ import {
...
 
+const SORT_STORAGE_KEY = 'hierarchy-sort';
+
+/** Read the persisted sort mode, falling back to the default for missing or
+ *  stale (e.g. renamed) localStorage values. */
+function readStoredSortMode(): HierarchySortMode {
+  if (typeof window === 'undefined') return DEFAULT_HIERARCHY_SORT;
+  const stored = localStorage.getItem(SORT_STORAGE_KEY);
+  return stored && (HIERARCHY_SORT_MODES as readonly string[]).includes(stored)
+    ? (stored as HierarchySortMode)
```
[the remark this answers is not in the document]
Comment (louistrue) [reply apps/viewer/src/components/viewer/hierarchy/useHierarchyTree.ts]: Fixed in db74c310. `localStorage` read and write are now wrapped in try/catch, falling back to the default mode and keeping the in-memory choice when storage is unavailable (private mode / quota).
